### PR TITLE
verify SSL

### PR DESCRIPTION
**Pull request by leoraba**
**Issue:** https://github.com/overture-stack/composer/issues/2

**Solution:**
- Configure `httpProxy` to use the `NODE_TLS_REJECT_UNAUTHORIZED` environment variable to disable certificate verification.
- Changed text to display when API token is encrypted by Keycloak.

### CODE REVIEW

File: components/pages/user/ApiTokenInfo.tsx [hunk 1]
```diff
@@ -284,6 +284,16 @@ const ApiTokenInfo = () => {
 	const parsedExpiry: number = existingApiToken ? parseExpiry(existingApiToken?.expiryDate) : 0;
 	const tokenIsExpired: boolean = has(existingApiToken, 'expiryDate') && parsedExpiry <= 0;
 
+	const displayToken = (token: ApiToken | null) => {
+		if (!token) {
+			return 'You have no API token...';
+		} else if (existingApiToken?.name) {
```
Comment (justincorrigible) [inline components/pages/user/ApiTokenInfo.tsx:290]: shouldn't this check be on `token` instead?

File: global/utils/proxyUtils.ts [hunk 2]
```diff
@@ -6,3 +6,7 @@ const convertToRegexPath = (path: string) => {
 export const removeFromPath = (url: string, path: string) => {
 	return url?.replace(convertToRegexPath(path), '') || '';
 };
+
+// Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections
+// and HTTPS requests insecure by disabling certificate verification
+export const SSLSecured = Number(process.env.NODE_TLS_REJECT_UNAUTHORIZED) != 0;
```
Comment (justincorrigible) [inline global/utils/proxyUtils.ts:12]: 🤔 A more strict check should prevent `false`, `Null` and empty strings being matched as if intending to disable this (as those values are [coerced to 0](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion)).

```suggestion
export const SSLSecured = !(Number(process.env.NODE_TLS_REJECT_UNAUTHORIZED) === 0);
```

File: components/pages/user/ApiTokenInfo.tsx (hunk 1 above)
Comment (justincorrigible) [reply components/pages/user/ApiTokenInfo.tsx:290]: In that case, we either remove EGO support (not doable right now) or acknowledge that the token *may* not include .name (via *optional* typing). That would then enable this...
```suggestion
		} else if (token?.name) {
```

File: pages/api/[...proxy].ts [hunk 3]
```diff
@@ -43,12 +43,11 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
 			changeOrigin: true,
+			secure: SSLSecured,
 		},
 		(err) => {
-			console.error(`Proxy error for ${req.url}: ${JSON.stringify(err)}`);
-			if (err) {
-				return res.status(500).json(err);
-			}
+			console.error(`Proxy error URL: ${req.url}. Error: ${JSON.stringify(err)}`);
+			return res.status(500).json(err);
```
Comment (justincorrigible) [inline pages/api/[...proxy].ts:50]:
```suggestion

			return res.status(500).json(err);
```

File: pages/api/protected/[...proxy].ts [hunk 4]
```diff
@@ -69,12 +69,11 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
 			changeOrigin: true,
+			secure: SSLSecured,
 		},
 		(err) => {
-			if (err) {
-				return res.status(500).end(err);
-			}
-			return res.end();
+			console.error(`Proxy error URL: ${req.url}. Error: ${JSON.stringify(err)}`);
+			return res.status(500).json(err);
```
Comment (justincorrigible) [inline pages/api/protected/[...proxy].ts:76]:
```suggestion

			return res.status(500).json(err);
```

File: components/pages/user/ApiTokenInfo.tsx (hunk 1 above)
Comment (leoraba) [reply components/pages/user/ApiTokenInfo.tsx]: updated `ApiToken` interface with optional `name?: string;`

**Pull request merged**